### DISCUSSION TITLE
How do I deal with personal projects for every user?

### DISCUSSION THREAD

**Discussion by stranljip** [kimai/kimai#3829] [Questions & Answers]
I want to provide a tool in my company to allow users to get rid of Toggl Track for tracking their work. At the moment a lot of people are using the Toggl Track possibility as a simple method to define their own projects and assign the time they worked on it to those issues. We currently have two tools which can extract the work logs from the Toggl API, deduce a Jira issue number from either the work log or from a project defined in Toggl and transfer the work log into Jira.
I already tried to use Kimai for this, mainly because I want to avoid having work log entries in a cloud service but I found that I cannot cover my use case with Kimai atm because there seems to be no possibility for a regular user to define personal projects. As far as I understand it, atm you need someone defining customers & projects in a central place and then providing those to the regular employees.
So - what I would like to have is the possibility for a user to define personal projects which are only visible to him/her. Is there anyone with a similar use case who can give me a hint how to implement this?

**Comment by HeinzWuert** [comment]
1. Assign the user the right to create projects via ROLES (I would like to create an complete new user)
2. After the project has been created, the user must be assigned to a team in which he is granted access rights and thus denied to other users.
That's it.

**Reply by stranljip** [reply]
Ok, so I need to create one team per user - that would be ok for me. I'll try that at the next possibility - TIA!

**Reply by HeinzWuert** [reply]
When the user creates the team he is automatically member of this team. The teamname is per default the name of the project.
When the users is not allowed to create teams the administration should create for every user  a team with the username and give them the right to edit teams of project.
Never used Kimai in this way but should work.

**Reply by kevinpapst** [reply]
There is even a setting for projects called "Take over teams from the logged-in user when creating new entries", which basically copies all teams from the creating user to the team.

That means as soon as the user is part of his own team and creates a project, only he will see it. No need to create a team per project.

**Reply by HeinzWuert** [reply]
That's new,,,nice!
But can't find it.
